enhancements/sdk-generate-commands.md

```diff
@@ -0,0 +1,411 @@
+generated with `--kustomize`, or creates a new one. This subcommand is interactive
+if a base does not exist.
+
+New project layout defaults:
+* Output directory: `config/bundle/manifests`
+* Read from: stdin, `--apis-dir="api(s)"`
+
+Old project layout defaults:
+* Output directory: `deploy/olm-catalog/<operator-name>/manifests`
+* Read from: `--manifest-root="deploy"`, `--crds-dir="deploy/crds"`, `--apis-dir="pkg/apis"`
+
+###### `--metadata`
+
+Generates a `bundle.Dockerfile` and `metadata/annotations.yaml`. Many flags exposed
+by [`opm alpha bundle generate`][opm-bundle-generate] will also be exposed by this
+step, and set by variables passed to `make`.
+
+New project layout defaults:
+* Output directory: `config/bundle/metadata`
+* Read from: _none_
+
+Old project layout defaults:
+* Output directory: `deploy/olm-catalog/<operator-name>/metadata`
+* Read from: _none_
+
+Notable flags: `--channels=[channel-names]`, `--default-channel=<channel-name>`, `--overwrite=<bool>`
+
+#### `genenerate packagemanifests`
+
+All files/manifests will be generated under `config/packages`, ex. `config/packages/0.0.1`.
+The functionality of this command is split up into steps by flag:
+`--kustomize` (for new project layouts only) and `--manifests`.
+
+###### `--kustomize` (new project layouts)
+
+Generate a `kustomize` base CSV for a new project layout under `config/packages/bases`.
+This base will be used as an actual base for `kustomize build` eventually; for now
+only `generate packagemanifests` will read this base. This step involves generating UI
+metadata from API types, although `--manifests` can do this as well. This subcommand
+is interactive if a base does not exist.
+
+New project layout defaults:
+* Output directory: `config/packages/bases`
+* Read from: `--apis-dir="api(s)"`
+
+###### `--manifests`
+
+Generate a package containing a CSV and owned CRD manifests, and updates the
+[package manifest file][package-manifest-doc]. This step uses a base generated
+with `--kustomize`, or creates a new one. This subcommand is interactive if a
+base does not exist.
+
+New project layout defaults:
+* Output directory: `config/packages/<version>`
+* Read from: stdin, `--apis-dir="api(s)"`
+
+Old project layout defaults:
+* Output directory: `deploy/olm-catalog/<operator-name>/<version>`
+* Read from: `--manifest-root="deploy"`, `--crds-dir="deploy/crds"`, `--apis-dir="pkg/apis"`
+
+Notable flags: `--version=<semver>`, `--channel=<channel-name>`, `--default-channel=<channel-name>`
+
+#### `make` recipes (new project layouts)
+
+New project layouts are scaffolded with a [Makefile][kb-makefile] containing a
+set of pre-defined recipes for generating manifests, building images, etc. To align
+with this idiom, recipes should be amended/added to include `generate bundle` by
+default. If a user does not want these features turned on in their project, they
+can manually remove the amendments/additions.
+
```

Comment:
I think we should provide the commands but we should NOT make mandatory its usage which would allow users to choose if they should or not use them as they wish.